components/funder/src/handler/handle_friend.rs

```diff
@@ -149,14 +144,7 @@ fn handle_request_send_funds<B>(
 ) where
     B: Clone + PartialEq + Eq + CanonicalSerialize + Debug,
 {
-    // Find ourselves on the route. If we are not there, abort.
-    let remote_index = request_send_funds
-        .route
-        .find_pk_pair(&remote_public_key, &m_state.state().local_public_key)
-        .unwrap();
-
-    let local_index = remote_index.checked_add(1).unwrap();
-    let next_index = local_index.checked_add(1).unwrap();
+    let next_index = 1;
```

Comment:
This line worries me. I thought it should be `= 0` but then it does not work. Is it correct that "destination" gets a non-empty route in this function?